### PR TITLE
Add ability to provide empty data to vectors layer

### PR DESCRIPTION
**Pull request by brisvag**
# Description
Unlike `Points`, `Image` and `Shapes`, `Vectors` currently fails if it receives `None` or empty data. This PR adds a `fix_data_vectors` function to mirror `fix_data_points`, which when possible/reasonable returns a valid Vectors data array.
Presumably the same can be done for the remaining layers.

This was prompted by an issue I had with a magicgui-based plugin, where I had to construct an empty Vectors layer and return that every time I had no data to provide.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: napari/_qt/layer_controls/qt_vectors_controls.py [hunk 1]
```diff
@@ -276,7 +276,10 @@ def _on_edge_color_change(self, event=None):
         event : napari.utils.event.Event, optional
             The napari event that triggered this method, by default None.
         """
-        if self.layer._edge.color_mode == ColorMode.DIRECT:
```
Comment (brisvag) [inline napari/_qt/layer_controls/qt_vectors_controls.py:279]: I'm not sure about this; it's deep in the weeds of a part of the codebase I never visited. Is it correct to assume that it's not a problem when there is no data to display?

**Review by brisvag — COMMENTED** [review]
Apart from the comment below, I have a general doubt: am I using the `ndim` kwarg correctly? It was passed to `Layer` with a hardcoded `2`, no matter the dimensionality...

File: napari/layers/vectors/vectors.py [hunk 2]
```diff
@@ -28,6 +32,9 @@ class Vectors(Layer):
         D dimensions. An (N1, N2, ..., ND, D) array is interpreted as
         "image-like" data where there is a length D vector of the
         projections at each pixel.
+    ndim : int
+        Number of dimensions for vectors. When data is not None, ndim must be D.
```
Comment (alisterburt) [inline napari/layers/vectors/vectors.py:32]:
```suggestion
        Number of dimensions for vectors. When data is not None, ndim defaults to D.
```

**Review by alisterburt — COMMENTED** [review]
At the minute `Vectors.ndim` must be `D` or `None`, maybe if `ndim` > `D` we should fill the extra dimensions with zeros?

File: napari/layers/vectors/vectors.py (hunk 2 above)
Comment (brisvag) [reply napari/layers/vectors/vectors.py:32]: I copied from the Points layer here... If the required `D` differs from the data, it won't change the data but simply raise an error.

Comment (alisterburt) [reply napari/layers/vectors/vectors.py:32]: okay cool, consistency is most important I guess!

Comment (brisvag) [reply napari/layers/vectors/vectors.py:32]: Though I agree that this might be a good idea... follow up PR maybe :)

Comment (brisvag) [reply napari/layers/vectors/vectors.py:32]: The more I think about this, the less I like it. I don't think there are any safe assumption to what a user would expect to happen when passing lowe dimensionality data. Prepend zeros if ndim is >3, but append them if it's lower (due to xyz ordering)? And there's probably a lot of cases where one would *want* the failure.

Comment (alisterburt) [reply napari/layers/vectors/vectors.py:32]: If we were to do it I would imagine following numpy broadcasting rules and prepending would be the way to go, but agree that erroring is the safest solution for now!

Comment (jni) [reply napari/layers/vectors/vectors.py:32]: > If we were to do it I would imagine following numpy broadcasting rules and prepending would be the way to go, but agree that erroring is the safest solution for now!

:100: 

@brisvag we don't do xyz. ;)

Comment (brisvag) [reply napari/layers/vectors/vectors.py:32]: Yeah, I agree, my point was that people *might* expect that, but get something different... In which case, I prefer the error :P

File: napari/_qt/layer_controls/qt_vectors_controls.py (hunk 1 above)
Comment (jni) [reply napari/_qt/layer_controls/qt_vectors_controls.py:279]: I think so.

File: napari/layers/vectors/vectors.py (hunk 2 above)
Comment (sofroniewn) [reply napari/layers/vectors/vectors.py:32]: Yup, i like this too

**Review by sofroniewn — APPROVED** [review]
LGTM! thanks for tests!! I will merge now

**Pull request merged**